torch/csrc/distributed/rpc/rref_proto.cpp

```diff
@@ -122,20 +100,32 @@ c10::intrusive_ptr<Message> RRefFetchRet::toMessageImpl() && {
   return fromIValues(values_, type_);
 }
 
+c10::intrusive_ptr<Message> ScriptRRefFetchRet::toMessageImpl() && {
+  auto res = fromIValues(values_, type_);
+  res->setDeviceMap(std::move(deviceMap_));
```

Comment:
curious, why do we set after creating the message instead of letting `fromIValues` handle `deviceMap_` as well?